0002/README.md

```diff
@@ -0,0 +1,209 @@
+provenance themselves.
+
+Issue: [#269](https://github.com/slsa-framework/slsa/issues/269)
+
+#### Milestone: SLSA policy model is decided, to detect or prevent publishing of packages containing unauthorized changes, with at least one representative implementation (Python and/or NPM)
+
+SLSA policies detect or prevent unauthorized changes for packages within the
+ecosystem. When opted-in, package maintainers must upload provenance alongside
+the package proving that the package was built directly from the canonical
+source repository. At first this may just display results on a dashboard,
+similar to [pyreadiness.org](https://pyreadiness.org/3.10/), but will hopefully
+be integrated directly into PyPI/NPM eventually. With this milestone, software
+producers can opt-in to automatically protect themselves against insider risk,
+and software consumers can gain assurance for opted-in packages, but the package
+registry is still trusted.
+
+Sub-problems:
+
+-   Policy model, which securely maps package name to canonical source
+    repository. For example, [requests](https://pypi.org/project/requests/) maps
+    to https://github.com/psf/requests. This metadata already exists for some
+    packages, but (a) it is not structured and (b) package maintainers can
+    mutate it at will. The solution must protect against malicious changes while
+    silently allowing official changes in source repository. One idea is to have
+    a Trust On First Use (TOFU) model with a mechanism for source repo
+    migrations.
+
+-   Implementation for each specific ecosystem, starting with Python/PyPI. This
+    includes a design, prototype, dashboard, and/or upload-time integration.
+
+-   (optional) Toy implementation of a mock package registry ("SLSA Playground")
+    to develop ideas and to show how SLSA would work in practice, without having
+    to change any existing system or to release any real package. This can also
+    be used as examples for https://slsa.dev/threats.
+
+#### Milestone: Client-side verification of SLSA policies in the Python ecosystem removes the need to trust the package registry
+
+SLSA verification happens in client tools that fetch Python packages, reducing
+or removing the need to trust the package registry (e.g. PyPI). This has a
+substantial reliability risk, so it is expected to come much later. Even in this
+model, the package registry enforces SLSA at upload time to prevent a maintainer
+from accidentally releasing a version that no one can use.
+
+This probably won't happen in 2022, but we list it here for completeness.
+
+### Theme: Adoption
+
+Widescale adoption of SLSA across open-source is the long-term goal, but it is
+not practical in 2022 because it not only depends on the work above, but also
+work to minimize or eliminate any friction for project maintainers.
+
+In the short term, we hope to have targeted adoption of key open-source projects
+to both exemplify standards and to reduce risk for those projects.
```

Comment:
An adoption milestone for this roadmap might be to share about SLSA and the work we're doing at appropriate forums for the representative implementation (i.e. WG's, conferences, etc).

We could also start capturing adoption stories (for many projects a "friends" repository has become a standard way to document adoption) – for example the recent Tekton case study.

Reply:
Added, though I'm not 100% confident in what I wrote. PTAL.

Reply:
LGTM, thanks!